### PR TITLE
Fix segfault for dense_sandwich

### PR DESCRIPTION
**Pull request by MartinStancsicsQC**
Checklist
* [x] Added a `CHANGELOG.rst` entry

### CODE REVIEW

**Comment by MartinStancsicsQC** [comment]
I think the deleted cast on L.229 was unnecessary, as `Cjj` is already `Py_ssize_t`, but it would be great is someone could confirm.

**Comment by jtilly** [comment]
The standard CI runners have 7GB of memory, that's probably not enough.

Please add the test to the test suite regardless and [add a marker](https://docs.pytest.org/en/7.1.x/example/markers.html). We won't run the test in CI (make sure to skip the marker [here](https://github.com/Quantco/tabmat/blob/main/.github/workflows/test.sh#L47)), but then it's straightforward to run it locally with `pytest tests`.

**Comment by MartinStancsicsQC** [comment]
Something like this?

File: CHANGELOG.rst [hunk 1]
```diff
@@ -7,6 +7,13 @@
 Changelog
 =========
 
+3.1.10 -- 2023-XX-XX
```
Comment (jtilly) [inline CHANGELOG.rst:10]:
```suggestion
3.1.10 -- 2023-06-23
```

**Review by jtilly — APPROVED** [review]

**Pull request merged**